widgetssdk/src/main/java/com/glia/widgets/callvisualizer/ActivityWatcherController.kt

```diff
@@ -172,18 +183,34 @@ internal class ActivityWatcherController(
                 exception?.run { watcher.showToast(exception.debugMessage) }
             }
 
-            override fun onScreenSharingStarted() {
+            override fun onScreenSharingRequestSuccess() {
                 // Should show screen sharing bubble
                 // Is handled by ActivityWatcherForChatHeadController
             }
+
+            override fun onScreenSharingStarted() {
+                permissionType?.run {
+                    if (this == OVERLAY) {
+                        currentDialogMode = MODE_OVERLAY_PERMISSION
+                        watcher.showOverlayPermissionsDialog()
+                        setOverlayPermissionRequestDialogShownUseCase.execute()
+                    }
+                }
+            }
+        }
+    }
+
+    @VisibleForTesting
+    internal fun onShowOverlayDialog(show: Boolean) {
+        if (show) {
+            this.permissionType = OVERLAY
         }
     }
```

Comment:
This function looks strange to me. Why?
1. It's not clear what it does, what is the purpose of it.
2. Name states that overlay dialog is showing, however, it accepts param that can be `false`. So... `onShowOverlayDialog` is called when dialog is not showing?
3. It's difficult to understand, what is the purpose of setting `OVERLAY` to `permissionType`.

Reply:
Removed with new logic

Reply:
> 1. It's not clear what it does, what is the purpose of it.
I think it was added for unit tests